Source/RunActivity/Viewer3D/Materials.cs

```diff
@@ -53,163 +63,97 @@ internal SharedTextureManager(Viewer viewer, GraphicsDevice graphicsDevice)
+                            // go up a level and try again. This repeats a fixed number of times, or until we run
+                            // out of known selector directories.
+                            var directory = Path.GetDirectoryName(levelPath);
+                            if (string.IsNullOrEmpty(directory) || !SelectorDirectoryNames.Contains(Path.GetFileName(directory))) break;
+                            levelPath = Path.Combine(Path.GetDirectoryName(directory), Path.GetFileName(levelPath));
                         }
+                        if (required) Trace.TraceWarning("Ignored missing texture file: {0}", path);
                     }
-                    else
+                    catch (Exception error)
                     {
-                        Trace.TraceWarning("Unsupported texture format: {0}", path);
-                        return defaultTexture;
+                        Trace.WriteLine(new FileLoadException(path, error));
                     }
-
-                    Textures.Add(path, texture);
-                    return texture;
-                }
-                catch (InvalidDataException error)
-                {
-                    Trace.TraceWarning("Skipped texture with error: {1} in {0}", path, error.Message);
                     return defaultTexture;
-                }
-                catch (Exception error)
-                {
-                    if (File.Exists(path))
-                        Trace.WriteLine(new FileLoadException(path, error));
-                    else
-                        Trace.TraceWarning("Ignored missing texture file {0}", path);
+                default:
+                    Trace.TraceWarning("Ignored unsupported texture file: {0}", path);
                     return defaultTexture;
-                }
-            }
-            else
-            {
-                return Textures[path];
             }
         }
 
-        public static Texture2D Get(GraphicsDevice graphicsDevice, string path)
+        // Internal callers expect a new `Texture2D` for every load so we must also provide a new missing texture for each.
+        internal static Texture2D GetInternalMissingTexture(GraphicsDevice graphicsDevice)
         {
-            if (path == null || path == "")
-                return SharedMaterialManager.MissingTexture;
-
-            path = path.ToLowerInvariant();
-            var ext = Path.GetExtension(path);
+            var texture = new Texture2D(graphicsDevice, 1, 1);
+            texture.SetData(new[] { Color.Magenta });
+            return texture;
+        }
 
-            if (ext == ".ace")
-                return Orts.Formats.Msts.AceFile.Texture2DFromFile(graphicsDevice, path);
-            else if (ext == ".dds" && File.Exists(path))
+        /// <summary>
+        /// Loads an internal texture file; DO NOT use with game data, use <see cref="Get(string, bool)"/> instead.
```

Comment:
Summary for LoadInternal() references Get() method.
Perhaps Get() method should reference LoadInternal() in a similar fashion?